gr-digital/python/digital/qa_mimo_channel_estimator_cc.py

```diff
@@ -0,0 +1,142 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with GNU Radio; see the file COPYING.  If not, write to
+# the Free Software Foundation, Inc., 51 Franklin Street,
+# Boston, MA 02110-1301, USA.
+# 
+
+from gnuradio import gr, gr_unittest
+from gnuradio import blocks
+import digital_swig as digital
+import numpy as np
+import pmt
+
+class qa_mimo_channel_estimator_cc (gr_unittest.TestCase):
+
+    def setUp (self):
+        self.tb = gr.top_block ()
+
+    def tearDown (self):
+        self.tb = None
+
+    def produce_training_seq(self, M, length):
+        seq = np.empty([M, length], dtype=complex)
+        for (m, i), value in np.ndenumerate(seq):
+            seq[m][i] = np.exp(-1j*2*np.pi*m*i/length)
+        return seq
+
+    def run_flowgraph(self, M, N, training_sequence, training_length, channel_matrix):
+        # Produce tags.
+        tags = [gr.tag_utils.python_to_tag((0,
+                                            pmt.string_to_symbol("pilot"),
+                                            pmt.from_long(0),
+                                            pmt.from_long(0)))]
+        # Build GNU Radio blocks.
+        data = np.random.randn(training_length) + 1j * np.random.randn(training_length)
+        src = [blocks.vector_source_c(data=np.append(training_sequence[0], data),
+                                      repeat=False,
+                                      tags=tags)]
+        sink = [blocks.vector_sink_c_make()]
+        for m in range(1, M):
+            src.append(blocks.vector_source_c(np.append(training_sequence[m], data)))
+        for n in range(1, N):
+            sink.append(blocks.null_sink_make(gr.sizeof_gr_complex))
+        # Use a matrix multiplier as a static channel model.
+        channel = blocks.multiply_matrix_cc_make(channel_matrix)
+        estimator = digital.mimo_channel_estimator_cc(M, N, training_sequence)
+        # Use a head block to terminate the flowgraph.
+        head = blocks.head_make(gr.sizeof_gr_complex, training_length)
+        # Connect everything.
+        for m in range(0, M):
+            self.tb.connect(src[m], (channel, m))
+        self.tb.connect((channel, 0), (estimator, 0), head, sink[0])
+        for n in range(1, N):
+            self.tb.connect((channel, n), (estimator, n), sink[n])
+        # Run flowgraph.
+        self.tb.run()
+        # Read channel estimation and write from pmt to numpy array.
+        csi = np.empty(shape=[N, M], dtype=complex)
+        for n in range(0, N):
+            for m in range(0, M):
+                csi[n][m] = pmt.c32vector_ref(pmt.vector_ref(sink[0].tags()[0].value, n), m)
+        return csi
+
+
+    '''
+    2 tests validating the correct estimation of the channel coefficients with a random channel
+    and 1xN MIMO scheme. '''
+    def test_001_t (self):
```

Comment:
This is fine, but for your own sanity in the future: test methods need to start with `test_` and can take names like `test_heavier_than_duck`.